src/extractor/extractor.cpp

```diff
@@ -115,142 +115,15 @@ transformTurnLaneMapIntoArrays(const guidance::LaneDescriptionMap &turn_lane_map
-        }
-        util::Log() << "timestamp: " << timestamp;
-
-        storage::io::FileWriter timestamp_file(config.timestamp_file_name,
-                                               storage::io::FileWriter::GenerateFingerprint);
-
-        timestamp_file.WriteFrom(timestamp.c_str(), timestamp.length());
-
-        // initialize vectors holding parsed objects
-        tbb::concurrent_vector<std::pair<std::size_t, ExtractionNode>> resulting_nodes;
-        tbb::concurrent_vector<std::pair<std::size_t, ExtractionWay>> resulting_ways;
-        tbb::concurrent_vector<boost::optional<InputRestrictionContainer>> resulting_restrictions;
-
-        // setup restriction parser
-        const RestrictionParser restriction_parser(scripting_environment);
-
-        // create a vector of iterators into the buffer
-        for (std::vector<osmium::memory::Buffer::const_iterator> osm_elements;
-             const osmium::memory::Buffer buffer = reader.read();
-             osm_elements.clear())
-        {
-            for (auto iter = std::begin(buffer), end = std::end(buffer); iter != end; ++iter)
-            {
-                osm_elements.push_back(iter);
-            }
-
-            // clear resulting vectors
-            resulting_nodes.clear();
-            resulting_ways.clear();
-            resulting_restrictions.clear();
-
-            scripting_environment.ProcessElements(osm_elements,
-                                                  restriction_parser,
-                                                  resulting_nodes,
-                                                  resulting_ways,
-                                                  resulting_restrictions);
-
-            number_of_nodes += resulting_nodes.size();
-            // put parsed objects thru extractor callbacks
-            for (const auto &result : resulting_nodes)
-            {
-                extractor_callbacks->ProcessNode(
-                    static_cast<const osmium::Node &>(*(osm_elements[result.first])),
-                    result.second);
-            }
-            number_of_ways += resulting_ways.size();
-            for (const auto &result : resulting_ways)
-            {
-                extractor_callbacks->ProcessWay(
-                    static_cast<const osmium::Way &>(*(osm_elements[result.first])), result.second);
-            }
-            number_of_relations += resulting_restrictions.size();
-            for (const auto &result : resulting_restrictions)
-            {
-                extractor_callbacks->ProcessRestriction(result);
-            }
-        }
-        TIMER_STOP(parsing);
-        util::Log() << "Parsing finished after " << TIMER_SEC(parsing) << " seconds";
-
-        util::Log() << "Raw input contains " << number_of_nodes << " nodes, " << number_of_ways
-                    << " ways, and " << number_of_relations << " relations";
+        auto turn_restrictions = ParseOSMData(scripting_environment, number_of_threads);
```

Comment:
Nice refactor. You could now also remove the scoping `{ }` since we only have the function scope now.